File: app/src/main/java/org/stepic/droid/store/dao/AssignmentDaoImpl.java
package org.stepic.droid.store.dao;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteOpenHelper;

import org.stepic.droid.model.Assignment;
import org.stepic.droid.store.structure.DbStructureAssignment;

public class AssignmentDaoImpl extends DaoBase<Assignment> {
    public AssignmentDaoImpl(SQLiteOpenHelper openHelper) {
        super(openHelper);
    }

    @Override
    public Assignment parsePersistentObject(Cursor cursor) {
        Assignment assignment = new Assignment();

        int columnIndexAssignmentId = cursor.getColumnIndex(DbStructureAssignment.Column.ASSIGNMENT_ID);
        int columnIndexCreateDate = cursor.getColumnIndex(DbStructureAssignment.Column.CREATE_DATE);
        int columnIndexProgress = cursor.getColumnIndex(DbStructureAssignment.Column.PROGRESS);
        int columnIndexStepId = cursor.getColumnIndex(DbStructureAssignment.Column.STEP_ID);
        int columnIndexUnitId = cursor.getColumnIndex(DbStructureAssignment.Column.UNIT_ID);
        int columnIndexUpdateDate = cursor.getColumnIndex(DbStructureAssignment.Column.UPDATE_DATE);

        assignment.setCreate_date(cursor.getString(columnIndexCreateDate));
        assignment.setId(cursor.getLong(columnIndexAssignmentId));
        assignment.setProgress(cursor.getString(columnIndexProgress));
        assignment.setStep(cursor.getLong(columnIndexStepId));
        assignment.setUnit(cursor.getLong(columnIndexUnitId));
        assignment.setUpdate_date(cursor.getString(columnIndexUpdateDate));
        return assignment;
    }

    @Override
    public String getDbName() {
        return DbStructureAssignment.ASSIGNMENTS;
    }

    @Override
    public ContentValues getContentValues(Assignment assignment) {
        ContentValues values = new ContentValues();

        values.put(DbStructureAssignment.Column.ASSIGNMENT_ID, assignment.getId());
        values.put(DbStructureAssignment.Column.CREATE_DATE, assignment.getCreate_date());
        values.put(DbStructureAssignment.Column.PROGRESS, assignment.getProgressId());
        values.put(DbStructureAssignment.Column.STEP_ID, assignment.getStep());
        values.put(DbStructureAssignment.Column.UNIT_ID, assignment.getUnit());
        values.put(DbStructureAssignment.Column.UPDATE_DATE, assignment.getUpdate_date());

        return values;
    }

    @Override
    public String getDefaultPrimaryColumn() {
        return DbStructureAssignment.Column.ASSIGNMENT_ID;
    }

    @Override
    public String getDefaultPrimaryValue(Assignment persistentObject) {
        return persistentObject.getId()+"";
    }
}

File: app/src/main/java/org/stepic/droid/storage/dao/AssignmentDaoImpl.kt
package org.stepic.droid.storage.dao

import android.content.ContentValues
import android.database.Cursor
import org.stepic.droid.storage.operations.DatabaseOperations
import org.stepic.droid.util.getDate
import org.stepic.droid.util.getLong
import org.stepic.droid.util.getString
import org.stepik.android.cache.assignment.structure.DbStructureAssignment
import org.stepik.android.model.Assignment
import javax.inject.Inject

class AssignmentDaoImpl
@Inject
constructor(
    databaseOperations: DatabaseOperations
) : DaoBase<Assignment>(databaseOperations) {
    public override fun getDbName(): String =
        DbStructureAssignment.TABLE_NAME

    public override fun parsePersistentObject(cursor: Cursor): Assignment =
        Assignment(
            id = cursor.getLong(DbStructureAssignment.Columns.ID),
            step = cursor.getLong(DbStructureAssignment.Columns.STEP),
            unit = cursor.getLong(DbStructureAssignment.Columns.UNIT),
            progress = cursor.getString(DbStructureAssignment.Columns.PROGRESS),

            createDate = cursor.getDate(DbStructureAssignment.Columns.CREATE_DATE),
            updateDate = cursor.getDate(DbStructureAssignment.Columns.UPDATE_DATE)
        )

    public override fun getContentValues(assignment: Assignment): ContentValues {
        val values = ContentValues()

        values.put(DbStructureAssignment.Columns.ID, assignment.id)
        values.put(DbStructureAssignment.Columns.PROGRESS, assignment.progress)
        values.put(DbStructureAssignment.Columns.STEP, assignment.step)
        values.put(DbStructureAssignment.Columns.UNIT, assignment.unit)
        values.put(DbStructureAssignment.Columns.CREATE_DATE, assignment.createDate?.time ?: -1)
        values.put(DbStructureAssignment.Columns.UPDATE_DATE, assignment.updateDate?.time ?: -1)

        return values
    }

    public override fun getDefaultPrimaryColumn(): String =
            DbStructureAssignment.Columns.ID

    public override fun getDefaultPrimaryValue(persistentObject: Assignment): String =
            persistentObject.id.toString()
}
